### PR TITLE
Tagging table

### PR DESCRIPTION
**Pull request by dtaylor7**
- add new tagging tables

### CODE REVIEW

File: models/tagEntity.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+TagEntity.init({
+  entityId: {
+    type: INTEGER,
+    field: "entity_id",
+  },
+  tagId: {
+    type: INTEGER,
+    field: "tag_id",
+  }
+}, { sequelize, modelName: 'tag', tableName: 'tag', timestamps: false });
```
Comment (mateen-hussain) [inline models/tagEntity.js:14]: `modelName` and `tableName ` can't be the same as `Tag` table, can they ?

File: migrations/28_add_tagging_tables.js [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+  try {
+    await query.dropTable('tag');
+  } catch (error) {
+    logger.error(`Error rolling back ${error}`);
+  }
+};
+
+module.exports = {
+  up: async (query) => {
```
Comment (mateen-hussain) [inline migrations/28_add_tagging_tables.js:56]: inconsistent pattern in defining `up` and `down` functions. Keep them them the same.

**Review by mateen-hussain — CHANGES_REQUESTED** [review]

File: migrations/28_add_tagging_tables.js (hunk 2 above)
Comment (dtaylor7) [reply migrations/28_add_tagging_tables.js:56]: not sure what you mean here?

Comment (dtaylor7) [reply migrations/28_add_tagging_tables.js:56]: actually i would rather stick with existing pattern, in many occasions the up function can be very large, for this reason keeps sense to keep execution and rollback separate

**Review by mateen-hussain — APPROVED** [review]

**Pull request merged**